tests/http/__init__.py

```diff
@@ -137,15 +138,28 @@ class TestServerTLSConnectionFactory:
     """An SSL connection creator which returns connections which present a certificate
     signed by our test CA."""
 
-    def __init__(self, sanlist):
+    def __init__(self, sanlist: List[bytes]):
         """
         Args:
-            sanlist: list[bytes]: a list of subjectAltName values for the cert
+            sanlist: a list of subjectAltName values for the cert
         """
         self._cert_file = create_test_cert_file(sanlist)
 
-    def serverConnectionForTLS(self, tlsProtocol):
+    def serverConnectionForTLS(self, tlsProtocol: TLSMemoryBIOProtocol) -> Connection:
         ctx = SSL.Context(SSL.SSLv23_METHOD)
         ctx.use_certificate_file(self._cert_file)
         ctx.use_privatekey_file(get_test_key_file())
         return Connection(ctx, None)
+
+
+@implementer(IAddress)
+class DummyAddress:
+    """Dummy stand-in for an IAddress.
+
+    Placates mypy. Useful for tests that don't care about addresses, but
+    about other HTTP machinery."""
+
+    pass
+
+
+dummy_address = DummyAddress()
```

Comment:
Should probably just use a dummy IPv4Address instance instead of this, c.f. [#14987 (comment)](https://github.com/matrix-org/synapse/pull/14987#discussion_r1096603919)

(reposting this comment after I force-pushed)